extmod/uasyncio/taskgroup.py

```diff
@@ -0,0 +1,244 @@
+# Adapted with permission from the EdgeDB project.
+# Adapted for MicroPython
```

Comment:
We need to be careful with licensing.  If this code was copied from CPython, that's possibly a show stopper.  The rest of uasyncio is MIT licensed and has original authorship specifically for MicroPython.

Reply:
The original is Apache licensed. That's sufficiently compatible with MIT (afaik) that I don't think we even need to ask for permission, though I intend to do that shortly anyway.